src/olmo_core/train/callbacks/list_checkpointer.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from dataclasses import dataclass, field
+from typing import List, Optional, Set
+
+from .checkpointer import CheckpointerCallback
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class ListCheckpointerCallback(CheckpointerCallback):
```

Comment:
This is a pretty strightforward (and useful) change, might be nicer to just implement it directly in CheckpointerCallback and that way we don't need to think about subclasses of CheckpointerCallback when we make modifications there.